e2e/e2e-suite_test.go

```diff
@@ -47,3 +51,18 @@ var _ = BeforeSuite(func() {
 	// parse test suite arguments
 	flag.Parse()
 })
+
+var _ = JustAfterEach(func() {
+	if CurrentSpecReport().Failed() {
+		failureLocation := CurrentSpecReport().Failure.Location
+		artifactsPath := filepath.Join(os.Getenv("ARTIFACTS"), fmt.Sprintf("%s:%d", filepath.Base(failureLocation.FileName), failureLocation.LineNumber))
+		cmd := exec.Command(filepath.Join(os.Getenv("BIN_DIR"), "dump"), "--kubeconfig", os.Getenv("KUBECONFIG"))
+		cmd.Env = append(cmd.Env, fmt.Sprintf("ARTIFACTS=%s", artifactsPath))
```

Comment:
any thoughts on how we could dump artifacts from the guest cluster as well? There are two layers we need to debug here. The infra cluster (which is what we have here) and the guest cluster (which may or may not exist at the time of failure)

Reply:
We are in control of when we dump, maybe we can override the kubeconfig and add something like "tenant-[clustername]" as directory prefix for the artifacts per test